pallets/market/src/lib.rs

```diff
@@ -268,9 +272,9 @@ pub mod pallet {
             )
         }
 
-        fn cid(&self) -> Result<Cid, ProposalError> {
+        fn cid(&self) -> Result<Cid, PaddedPieceSizeError> {
```

Comment:
There is no need for this function. Also the error type is confusing. Use `Commitment::from_cid_bytes`. It does the conversion and it also checks if cid is correct. Hmm. If you want to have a function on the struct then it would be better to have something like `fn piece_commitment(&self) -> Commitment<CommP>`. That is because the piece cid actually encodes the commitment.

Now that I think about it. The `piece_commitment` function is better. That way we are ready if/when we implement TypeInfo for the `Commitment` struct and use that directly on the runtime struct.